styx-common/src/main/java/com/spotify/styx/storage/DatastoreStorage.java

```diff
@@ -308,6 +320,39 @@ public void updateNextNaturalTrigger(WorkflowId workflowId, TriggerInstantSpec t
     return map;
   }
 
+  public Map<WorkflowId, Workflow> workflows(Set<WorkflowId> workflowIds) {
+    final Callable<Map<WorkflowId, Workflow>> callable =
+        () -> Lists
+            .partition(ImmutableList.copyOf(workflowIds), MAX_NUMBER_OF_ENTITIES_IN_ONE_BATCH)
+            .parallelStream()
+            .map(this::getBatchOfWorkflows)
+            .flatMap(Collection::stream)
+            .collect(toMap(Map.Entry::getKey, Map.Entry::getValue));
+    return forkJoinPool.submit(callable).join();
+  }
+
+  private Set<Map.Entry<WorkflowId, Workflow>> getBatchOfWorkflows(final List<WorkflowId> batch) {
+    final Set<Key> keys = batch.stream()
```

Comment:
There's a lot of seemingly unnecessary work here in creating maps / sets and transforming them to sequential streams and then repackaging them as maps again.